trio/_util.py

```diff
@@ -183,11 +199,11 @@ class ConflictDetector:
 
     """
 
-    def __init__(self, msg):
+    def __init__(self, msg: str):
```

Comment:
Add ` -> None`

Reply:
It's not actually required. Type checkers know that `__init__` will always return `None`, so it's fine to leave it unannotated as long as at least one parameter is annotated to prevent the gradual typing logic from just skipping the method.

Reply:
Mypy complains if you up the strictness I'm pretty sure

Reply:
Took a bit of searching, but found this [Mypy code](https://github.com/python/mypy/blob/002502a0111852c360f2255830951473bcfec4a7/mypy/semanal.py#L850-L853) which converts `-> Any` annotations to `-> None` for `__init__`. There's also a specific [error](https://github.com/python/mypy/blob/002502a0111852c360f2255830951473bcfec4a7/mypy/message_registry.py#L84) for not returning `None`.

Reply:
I don't have strong opinions either way, but if we want to enforce that it should be enabled - otherwise it's covered by `no_untyped_defs` only enforcing `-> None` on `__init__` without typed params.